File: Valghalla.Internal.Web/src/features/administration/election/models/duplicate-election-request.ts
export interface DuplicateElectionTaskTypeCommunicationTemplateRequest {
  taskTypeId: string;
  confirmationOfRegistrationCommunicationTemplateId?: string;
  confirmationOfCancellationCommunicationTemplateId?: string;
  invitationCommunicationTemplateId?: string;
  invitationReminderCommunicationTemplateId?: string;
  taskReminderCommunicationTemplateId?: string;
  retractedInvitationCommunicationTemplateId?: string;
  removedFromTaskCommunicationTemplateId?: string;
  removedByValidationCommunicationTemplateId?: string;
}

export interface DuplicateElectionRequest {
  sourceElectionId: string;
  title: string;
  electionTypeId: string;
  lockPeriod: number;
  electionDate: Date;
  daysBeforeElectionDate: number;
  daysAfterElectionDate: number;  
  workLocationIds: string[];
  confirmationOfRegistrationCommunicationTemplateId?: string;
  confirmationOfCancellationCommunicationTemplateId?: string;
  invitationCommunicationTemplateId?: string;
  invitationReminderCommunicationTemplateId?: string;
  taskReminderCommunicationTemplateId?: string;
  retractedInvitationCommunicationTemplateId?: string;
  removedFromTaskCommunicationTemplateId?: string;
  removedByValidationCommunicationTemplateId?: string;
  electionTaskTypeCommunicationTemplates?: Array<DuplicateElectionTaskTypeCommunicationTemplateRequest>;
}

File: Valghalla.Internal.API/Requests/Administration/Election/DuplicateElectionRequest.cs
using Valghalla.Internal.Application.Modules.Administration.Election.Requests;

namespace Valghalla.Internal.API.Requests.Administration.Election
{
    public sealed record DuplicateElectionRequest
    {
        public Guid SourceElectionId { get; init; }
        public string Title { get; init; } = string.Empty;
        public Guid ElectionTypeId { get; init; }
        public int LockPeriod { get; init; }
        public DateTime ElectionDate { get; init; }
        public int DaysBeforeElectionDate { get; init; }
        public int DaysAfterElectionDate { get; init; }
        public Guid? ConfirmationOfRegistrationCommunicationTemplateId { get; set; }
        public Guid? ConfirmationOfCancellationCommunicationTemplateId { get; set; }
        public Guid? InvitationCommunicationTemplateId { get; set; }
        public Guid? InvitationReminderCommunicationTemplateId { get; set; }
        public Guid? TaskReminderCommunicationTemplateId { get; set; }
        public Guid? RetractedInvitationCommunicationTemplateId { get; set; }
        public Guid? RemovedFromTaskCommunicationTemplateId { get; set; }
        public Guid? RemovedByValidationCommunicationTemplateId { get; set; }
        public IEnumerable<Guid> WorkLocationIds { get; init; } = Array.Empty<Guid>();
        public IEnumerable<CreateElectionTaskTypeCommunicationTemplateRequest> ElectionTaskTypeCommunicationTemplates { get; init; } = Array.Empty<CreateElectionTaskTypeCommunicationTemplateRequest>();
    }
}
